src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -0,0 +1,813 @@
+                        while (!string.IsNullOrWhiteSpace(await serverReader1.ReadLineAsync()));
+                        await serverWriter1.WriteAsync(responseBody);
+                        await request1;
+
+                        // Make second request and expect it to be served from a different connection.
+                        Task<string> request2 = client.GetStringAsync(uri);
+                        await LoopbackServer.AcceptSocketAsync(listener, async (server2, serverStream2, serverReader2, serverWriter2) =>
+                        {
+                            while (!string.IsNullOrWhiteSpace(await serverReader2.ReadLineAsync()));
+                            await serverWriter2.WriteAsync(responseBody);
+                            await request2;
+                            return null;
+                        });
+
+                        return null;
+                    });
+                });
+            }
+        }
+
+        [Theory]
+        [InlineData("PooledConnectionLifetime")]
+        [InlineData("PooledConnectionIdleTimeout")]
+        public async Task SmallConnectionTimeout_SubsequentRequestUsesDifferentConnection(string timeoutPropertyName)
+        {
+            using (var handler = new SocketsHttpHandler())
+            {
+                switch (timeoutPropertyName)
+                {
+                    case "PooledConnectionLifetime": handler.PooledConnectionLifetime = TimeSpan.FromMilliseconds(1); break;
+                    case "PooledConnectionIdleTimeout": handler.PooledConnectionLifetime = TimeSpan.FromMilliseconds(1); break;
+                    default: throw new ArgumentOutOfRangeException(nameof(timeoutPropertyName));
+                }
+
+                using (HttpClient client = new HttpClient(handler))
+                {
+                    await LoopbackServer.CreateServerAsync(async (listener, uri) =>
+                    {
+                        // Make first request.
+                        Task<string> request1 = client.GetStringAsync(uri);
+                        await LoopbackServer.AcceptSocketAsync(listener, async (server1, serverStream1, serverReader1, serverWriter1) =>
+                        {
+                            while (!string.IsNullOrWhiteSpace(await serverReader1.ReadLineAsync()));
+                            await serverWriter1.WriteAsync(LoopbackServer.DefaultHttpResponse);
+                            await request1;
+
+                            // Wait a small amount of time before making the second request, to give the first request time to timeout.
+                            await Task.Delay(100);
+
+                            // Make second request and expect it to be served from a different connection.
+                            Task<string> request2 = client.GetStringAsync(uri);
+                            await LoopbackServer.AcceptSocketAsync(listener, async (server2, serverStream2, serverReader2, serverWriter2) =>
```

Comment:
>AcceptSocketAsync [](start = 49, length = 17)

Q: this is a pattern in most of the tests, but in case of client/handler bug the call to AcceptSocketAsync can cause the test to hang. Is there any reason to not add a timeout on the `AcceptAsync()` call  on `LoopbackServer.AcceptSocketAsync`?

Reply:
We can do that, we would just want it to be really long, like a minute.